### PR TITLE
Sorting dijets by both cmva and deepCSV

### PR DESCRIPTION
**Pull request by alesaggio**

### CODE REVIEW

File: plugins/Categories.cc [hunk 1]
```diff
@@ -48,12 +53,16 @@ bool MuMuCategory::event_in_category_pre_analyzers(const ProducersManager& produ
...
 
     if (ll.empty())
         return false;
 
-    if (lljj.empty())
+    if (lljj_cmva.empty())
+        return false;
+    
+    if (lljj_deepCSV.empty())
         return false;
```
Comment (OlivierBondu) [inline plugins/Categories.cc]: wait... this I did not foresee: it means that the category will be determined by one or the other btag algo ?
The lepton pair selected should not change given we have no criteria to select the lljj candidate besides the keeping the only the one with the largest btag sum

so : shouldn't this be `if (lljj_cmva.empty() && lljj_deepCSV.empty())` instead ? I think the rate of false positive is exactly 0

File: plugins/HtoZAAnalyzer.cc [hunk 2]
```diff
@@ -732,9 +739,16 @@ void HtoZAAnalyzer::analyze(const edm::Event& event, const edm::EventSetup&, con
...
+    for (unsigned int ijet=0; ijet < jets.size(); ijet++) {
+        HT += jets[ijet].p4.Pt();
+    }
+    
+    if (lljj_deepCSV.size() > 0)
+        // take the first lljj_cmva since it's been resized: it has only one entry
+        HT += lljj_deepCSV[0].lep1_p4.Pt() + lljj_deepCSV[0].lep2_p4.Pt();
     for (unsigned int ijet=0; ijet < jets.size(); ijet++) {
         HT += jets[ijet].p4.Pt();
     }
```
Comment (OlivierBondu) [inline plugins/HtoZAAnalyzer.cc]: This doesn't work, there should be two `HT` variables depending on the btag algo and hence on the jets picked...

**Review by OlivierBondu — COMMENTED** [review]
Thanks! I see only two errors in the logic, I take it you have checked this work ?

File: plugins/Categories.cc (hunk 1 above)
Comment (alesaggio) [reply plugins/Categories.cc]: Yes, you are completely right, the category cannot be btagger-dependent. I'll fix it ;)

File: plugins/HtoZAAnalyzer.cc (hunk 2 above)
Comment (alesaggio) [reply plugins/HtoZAAnalyzer.cc]: Ah yes sorry, I missed it. Fixiiing

**Comment by alesaggio** [comment]
Updated. Sorry for the delay, I wanted to make sure that everything made sense ;)

**Comment by OlivierBondu** [comment]
Thanks ! 

although... there are 4 categories in total, the other 3 still need fixing :wink:

**Comment by alesaggio** [comment]
Done!

**Comment by OlivierBondu** [comment]
Thanks!

**Pull request merged**